### PR TITLE
Update  `gotools` and `amptools` images versioning

### LINKED ISSUE #1692: Update `gotools` and `amptools` images to use fixed versions

### PR DESCRIPTION
**Pull request by neha-viswanathan**
Fixes #1692 

This PR has the following changes:
- Update versions of some of the packages in `gotools` image
- `install-protoc` now downloads the binary instead of building from source
- Bump the version of `gotools` image to `1.12.0` and `amptools` image to `1.9.0`
- Update `ampmake` binaries

#### To test
```
$ cd images/gotools
$ ./build
$ cd ../amptools
$ ./build
$ cd ../..
$ ampmake clean build
```

### CODE REVIEW

**Review by bquenin — DISMISSED** [review]
LGTM (tested on linux)

File: images/gotools/bin/build-protoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (ndegory) [inline images/gotools/bin/build-protoc:1]: is this file needed? I don't see any reference elsewhere.

File: images/gotools/bin/install-protoc [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
-cd ..
-rm -r protobuf
-/usr/local/bin/protoc --version
+VERSION=3.4.0
+wget -N -q "https://github.com/google/protobuf/releases/download/v$VERSION/protoc-$VERSION-linux-x86_64.zip" 2>&1
+unzip -q "protoc-$VERSION-linux-x86_64.zip"
+mv bin/protoc /usr/local/bin
+chmod +x /usr/local/bin/protoc
+mv include/google /usr/local/include/google
+chmod -R 655 /usr/local/include/google
```
Comment (ndegory) [inline images/gotools/bin/install-protoc]: 655 should probably be 755

**Review by ndegory — CHANGES_REQUESTED** [review]

File: images/gotools/bin/install-protoc (hunk 2 above)
Comment (neha-viswanathan) [reply images/gotools/bin/install-protoc]: Since these are just `include` files (not binaries), they do not require the executable flag.

File: images/gotools/bin/build-protoc (hunk 1 above)
Comment (neha-viswanathan) [reply images/gotools/bin/build-protoc:1]: This file has been retained in case we will revert back to using the alpine version of the go image (since there are no pre-compiled binaries available for that system).

File: images/gotools/bin/install-protoc (hunk 2 above)
Comment (neha-viswanathan) [reply images/gotools/bin/install-protoc]: ok will change it.

**Review by ndegory — APPROVED** [review]

**Review by bquenin — APPROVED** [review]
LGTM (tested on linux & windows)
